react-components/src/architecture/base/renderStyles/DepthTestRenderStyle.ts

```diff
@@ -0,0 +1,13 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { RenderStyle } from './RenderStyle';
+
+export abstract class DepthTestRenderStyle extends RenderStyle {
+  // ==================================================
+  // INSTANCE FIELDS
+  // ==================================================
+
+  public depthTest = true;
+}
```

Comment:
Is this a necessary abstraction? I'm just imagining if we hypothetically want e.g. a "isClipped" renderStyle and create a `IsClippedRenderStyle` abstract class, it suddenly becomes impossible to inherit from both `DepthTestRenderStyle` and `IsClippedRenderStyle` simultaneously if we want to use both in the same class. We could get around this by using type unions (`type NewObjectRenderStyle = DepthTestRenderStyle & IsClippedRenderStyle`), but it might also have some drawbacks.

Reply:
- From now you can use use a proper IDE for PR review. If you search from DepthTestRenderStyle you would understand why. And Yes, you point out a problem in Object Orientated Programming, but this is rare and more a corner case. One way is to nest the render style into for instance one LineStyle, one SolidStyle and one LabelStyle, as I have done in the debugger code. Then you can combine all, and use OO at the same time. Using types instead ? Why? haven't you done that in Reveal over a long term? I haven't seen them as part of any useful architecture there.

Reply:
So I called this CommonRenderStyle instead. So if you need, you can add other common tings here. You do not need to use all fields in all domain objects.